docs/reference/query-dsl.asciidoc

```diff
@@ -25,6 +25,27 @@ or to alter their behaviour (such as the
 
 Query clauses behave differently depending on whether they are used in
 <<query-filter-context,query context or filter context>>.
+
+[[query-dsl-allow-expensive-queries]]
+Allow expensive queries::
+Execution of certain types of queries have usually slow performance, which can affect the cluster performance.
```

Comment:
Certain types of queries will generally execute slowly due to the way they are implemented, which can affect the stability of your cluster.